core-java-modules/core-java-nio-2/src/main/java/com/baeldung/lock/FileLocks.java

```diff
@@ -0,0 +1,239 @@
+package com.baeldung.lock;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.nio.ByteBuffer;
+import java.nio.channels.FileChannel;
+import java.nio.channels.FileLock;
+import java.nio.channels.NonReadableChannelException;
+import java.nio.channels.NonWritableChannelException;
+import java.nio.channels.OverlappingFileLockException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Charsets;
+
+
+public class FileLocks {
+	
+	private static Logger log = LoggerFactory.getLogger(FileLocks.class);
+
+	// Write locks
+	/**
+	 * Trying to get an exclusive lock on a read-only FileChannel won't work.
+	 */
+	static void getExclusiveLockFromInputStream() throws IOException, NonWritableChannelException {
+		Path path = Files.createTempFile("foo", "txt");
+		try (FileInputStream fis = new FileInputStream(path.toFile()); FileLock lock = fis.getChannel().lock()) {
+			log.debug("This won't happen");
+		} catch (NonWritableChannelException e) {
+			log.error(
+					"The channel obtained through a FileInputStream isn't writable. "
+					+ "You can't obtain an exclusive lock on it!");
+			throw e;
+		}
+	}
+
+	
+	/**
+	 * Getting an exclusive lock from a RandomAccessFile works if the file is in write mode.
+	 * @param from beginning of the locked region
+	 * @param size how many bytes to lock
+	 * @return
+	 * @throws IOException
+	 */
+	static FileLock getExclusiveLockFromRandomAccessFile(long from, long size) throws IOException {
+		Path path = Files.createTempFile("foo", "txt");
+		try (RandomAccessFile file = new RandomAccessFile(path.toFile(), "rw");
+				FileLock lock = file.getChannel().lock(from, size, false)) {
+			if (lock.isValid()) {
+				log.debug("This is a valid exclusive lock");
+				return lock;
+			}
+			return null;
+		} catch (Exception e) {
+			System.out.println(e.getMessage());
+		}
+		return null;
+	}
+	
+	/**
+	 * Getting a write lock on a file region
+	 */
+	static FileLock getExclusiveLockFromFileChannelOpen(long from, long size) throws IOException {
+		Path path = Files.createTempFile("foo", "txt");
+		try (FileChannel channel = FileChannel.open(path, StandardOpenOption.APPEND);
+				FileLock lock = channel.lock(from, size, false)) {
+			String text = "Hello, world.";
+			ByteBuffer buffer = ByteBuffer.allocate(text.length() + System.lineSeparator().length());
+			buffer.put((text + System.lineSeparator()).getBytes(Charsets.UTF_8));
```

Comment:
Charsets is a guava class, was that deliberate? as this article is about NIO I think it would be better to use..

- StandardCharsets.UTF_8